packages/synapse-react-client/src/components/DataGrid/SynapseGrid.tsx

```diff
@@ -209,9 +214,27 @@ const SynapseGrid = forwardRef<
       }
 
       if (operation.type === 'UPDATE') {
```

Comment:
Why is `handleChange` being called on scroll? I think adding this check is OK, but this check could slow down the browser if it's checking each row as you scroll past it